### ISSUE TITLE
SFTP Adapter doesn't delete directory

### ISSUE THREAD

**Issue by Naoray** [thephpleague/flysystem#1439]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 3.0.7

#### Summary
When attempting to delete a via sftp created empty directory, the directory is not deleted but the `deleteDirectory()` method marks the operation as successful by returning `true`.

#### How to reproduce

```php
$flysystem = new Filesystem(new SftpAdapter(...));
$config = ['visibility' => 'public'];

$flysystem->createDirectory('foo', $config);

$flysystem->deleteDirectory('foo');

// Not sure if the code above actually works
// here is the code I use inside my Laravel project
Storage::disk('sftp')->createDirectory('foo');
Storage::disk('sftp')->deleteDirectory('foo');
```

Based on [this comment in a phpseclib issue](https://github.com/phpseclib/phpseclib/issues/1677#issuecomment-870501166) I modified the call to the phpseclib to use `rmdir()` instead of `delete()`. After that change my directory was deleted successfully

```diff
- $connection->delete(rtrim($location, '/') . '/');
+ $connection->rmdir(rtrim($location, '/') . '/');
```

I did try to do a PR, but changing this to `rmdir()` does seem to make other tests fail (s. https://github.com/Naoray/flysystem/actions/runs/2091041924)

**Comment by frankdejonge** [comment]
I think I understand why your tests fail. Another comment on that page indicates that the delete removes everything inside the directory, except for the top-level directory. The rmdir function only works if the directory is empty. This means both need to be called, with the rmdir after the delete call.

**Comment by frankdejonge** [comment]
Fixed in 3.0.15
